src/skmultiflow/meta/streaming_random_patches.py

```diff
@@ -0,0 +1,667 @@
+            else:
+                k = self._random_state.poisson(lam=self.lam)
+                if k > 0:
+                    self.ensemble[i].partial_fit(X=X, y=y, classes=classes,
+                                                 sample_weight=np.asarray([k]),
+                                                 n_samples_seen=self._n_samples_seen,
+                                                 random_state=self._random_state)
+
+    def predict(self, X):
+        """ Predict classes for the passed data.
+
+        Parameters
+        ----------
+        X : numpy.ndarray of shape (n_samples, n_features)
+            The set of data samples to predict the class labels for.
+
+        Returns
+        -------
+        A numpy.ndarray with all the predictions for the samples in X.
+
+        """
+        n_samples, n_features = X.shape
+
+        if self.ensemble is None:
+            self._init_ensemble(n_features=n_features)
+            return np.zeros(n_samples)
+
+        y_proba = self.predict_proba(X)
+        y_pred = np.argmax(y_proba, axis=1)
+        return y_pred
+
+    def predict_proba(self, X):
+        """ Estimate the probability of X belonging to each class-labels.
+
+        Parameters
+        ----------
+        X : numpy.ndarray of shape (n_samples, n_features)
+            Samples one wants to predict the class probabilities for.
+
+        Returns
+        -------
+        A numpy.ndarray of shape (n_samples, n_labels), in which each outer
+        entry is associated with the X entry of the same index. And where the
+        list in index [i] contains len(self.target_values) elements, each of
+        which represents the probability that the i-th sample of X belongs to
+        a certain class-label.
+
+        """
+        n_samples, n_features = get_dimensions(X)
+        y_proba = []
+
+        if self.ensemble is None:
+            self._init_ensemble(n_features=n_features)
+            return np.zeros(n_samples)
+
+        for i in range(n_samples):
+            y_proba.append(self._predict_proba(np.asarray([X[i]])))
+        return np.asarray(y_proba)
+
+    def _predict_proba(self, X):
+        y_proba = np.asarray([0.])
+
+        for i in range(len(self.ensemble)):
+            y_proba_temp = self.ensemble[i].predict_proba(X)
+            if np.sum(y_proba_temp) > 0.0:
+                y_proba_temp = normalize(y_proba_temp, norm='l1')[0].copy()
+                acc = self.ensemble[i].performance_evaluator.accuracy_score()
+                if not self.disable_weighted_vote and acc > 0.0:
+                    y_proba_temp *= acc
+                # Check array length consistency
+                if len(y_proba_temp) != len(y_proba):
+                    if len(y_proba_temp) > len(y_proba):
+                        y_proba.resize((len(y_proba_temp), ), refcheck=False)
+                    else:
+                        y_proba_temp.resize((len(y_proba), ), refcheck=False)
```

Comment:
The base learner in the `AdaptiveRandomForestClassifier` is the `HoeffdingTreeClassifier` which can take `classes` to set the size of the probabilities array returned. In the `StreamingRandomPatchesClassifier` the base learner can be any type of estimator, so it can use an estimator that does not take into account `classes`.

Reply:
Nice catch. Would in this case the probability vectors might become "unaligned"? (In case the base models observed different subsets of the classes)

Reply:
The intention of the `resize` operation is to account for this case. For both cases, differences in other members of the ensemble, or in the current estimator.

Reply:
Thanks for the further clarification!